interpreter/src/test/java/tw/waterball/survive/io/interpreter/impl/StreamLiteralReaderTest.java

```diff
@@ -0,0 +1,105 @@
+package tw.waterball.survive.io.interpreter.impl;
+
+import org.junit.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.EOFException;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+import static org.junit.Assert.*;
+
+/**
+ * TODO:
+ *  - column / row / position calculation isn't strictly tested and implemented
+ * @author Waterball (johnny850807@gmail.com)
+ */
+public class StreamLiteralReaderTest {
+    private StreamLiteralReader literalReader;
+
+    @Test
+    public void givenText_Apple_whenNextChar3Times_shouldBe_A_p_p_respectively() throws IOException {
+        givenText("Apple");
+
+        assertSame('A', literalReader.nextChar());
+        assertSame('p', literalReader.nextChar());
+        assertSame('p', literalReader.nextChar());
+    }
+
+    @Test
+    public void givenText_Apple_goToPos3_whenGetCurrentChar_shouldBe_p() throws IOException {
+        givenText("Apple");
+        goToPosition(3);
+        assertSame('p', literalReader.getCurrentChar());
+    }
+
+
+    @Test
+    public void givenText_Apple_goToPos3_whenGetCurrentPosition_shouldBe3() throws IOException {
+        givenText("Apple");
+        goToPosition(3);
+
+        assertSame(3, literalReader.getCurrentPosition());
+    }
+
+    @Test
+    public void givenText_Apple_goToPos3_whenGetCurrentColumn_shouldBe3() throws IOException {
+        givenText("Apple");
+        goToPosition(3);
+
+        assertSame(3, literalReader.getCurrentColumn());
+    }
+
+    @Test
+    public void givenText_Apple_goToPos3_whenGetCurrentRow_shouldBe0() throws IOException {
+        givenText("Apple");
+        goToPosition(3);
+
+        assertSame(0, literalReader.getCurrentRow());
```

Comment:
From the idea that `pos` is starting from 1, I would expect that the `row` should start from 1...

Reply:
From my above explanation, everything starts from 0 exactly.